internal/db/pg/pets.go

```diff
@@ -27,3 +27,51 @@ func (r *Repository) CreatePet(ctx context.Context, p *gostreamv1.Pet) (uint64,
 
 	return p.Id, nil
 }
+
+// ListPets ...
+func (r *Repository) ListPets(ctx context.Context) ([]*gostreamv1.Pet, error) {
+	q := sq.Select("id", "kind", "name", "age").From("pets").
+		PlaceholderFormat(sq.Dollar)
+
+	query, params, err := q.ToSql()
+	if err != nil {
+		fmt.Println("sq error: ", err)
+	}
+	rows, err := r.pool.Query(ctx, query, params...)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	defer rows.Close()
+	result := make([]*gostreamv1.Pet, 0)
+	for rows.Next() {
+		var u gostreamv1.Pet
+		if err = rows.Scan(&u.Id, &u.Kind, &u.Name, &u.Age); err != nil {
+			fmt.Println(err)
+		}
+
+		result = append(result, &u)
+	}
+	return result, err
+}
+
+// UpdatePet ...
+func (r *Repository) UpdatePet(ctx context.Context, p *gostreamv1.Pet) error {
+	id := p.GetId()
+	q := sq.Update("pets").Where("id = ?", id).
+		PlaceholderFormat(sq.Dollar)
+	q = q.Set("kind", p.GetKind())
+	q = q.Set("name", p.GetName())
+	q = q.Set("age", p.GetAge())
```

Comment:
Not so important but you can chain the queries like the following:

```go
q := sq.Update("pets").Where("id = ?", id).
		PlaceholderFormat(sq.Dollar).
                 Set("kind", p.GetKind()).
                 Set("name", p.GetName()).
	        Set("age", p.GetAge())
```

Reply:
done